external/pmrrr/CMakeLists.txt

```diff
@@ -10,10 +10,13 @@
 #  http://opensource.org/licenses/BSD-2-Clause
 #
 
+# Create a set of defines for including in templated code
+set(pmrrr_defines "\n")
+
 option(HAVE_SPINLOCKS "Enable if pthread lib supports spinlocks" OFF)
 MARK_AS_ADVANCED(HAVE_SPINLOCKS)
 if(NOT HAVE_SPINLOCKS)
-  add_definitions(-DNOSPINLOCKS)
+  set(pmrrr_defines "${pmrrr_defines}#define NOSPINLOCKS\n")
```

Comment:
Thank you for the detailed reply; though it seems it would be more usual to use `cmakedefine` within the configure file rather than explicit string includes.